docs/concepts/controller-manager.md

```diff
@@ -107,6 +107,25 @@ The `--event-ttl` should be larger than the `ttlNonShootEvents` or this controll
 
 Consequently, to ensure that `ExposureClass`s in-use are always present in the system until the last referring `Shoot` gets deleted, the controller adds a finalizer which is only released when there is no `Shoot` referencing the `ExposureClass` anymore.
 
+### [`ManagedSeedSet` Controller](../../pkg/controllermanager/controller/managedseedset)
+
+`ManagedSeedSet` objects maintain a stable set of replica of `ManagedSeed`s, i.e they guarantee the availability of a specified number of identical `ManagedSeed`s on an equal number of identical `Shoot`s.
+`ManagedSeedSet` controller creates and deletes `ManagedSeed`s and `Shoot`s in response to changes to the replicas and selector fields. For more info refer to the [`ManagedSeedSet` proposal document](../../docs/proposals/13-automated-seed-management.md#managedseedsets).
+
+1. The reconciler first gets all the replicas of the given `ManagedSeedSet` in the `ManagedSeedSet`'s namespace and with the matching selector. Each replica is a tuple that contains a `ManagedSeed`, its corresponding `Seed` and `Shoot` objects.
+1. Then the pending replica is retrieved, if it exists.
+1. Next it determines the ready, postponed and deletable replica.
+    - A replica is considered `ready` when a `Seed` owned by a `ManagedSeed` has been registered either directly or by deploying `gardenlet` into a `Shoot`, the `Seed` is `Ready` and the `Shoot`'s status is `Healthy`.
+    - If a replica is not ready and it is not pending, i.e it is not specified in the `ManagedSeed`'s `status.pendingReplica` field, then it is added to the `postponed` replicas.
+    - A replica is deletable if it has no scheduled `Shoot`s and is not protected by the `protect-from-deletion` annotation.
```

Comment:
> by the `protect-from-deletion` annotation.

What's that? Is that something specific to this controller? Maybe introduce this somewhere above.

Reply:
yes a replica is marked as deletable when shoot and managedseed of the replica does not have `seedmanagement.gardener.cloud/protect-from-deletion` annotation present and if there is no shoot scheduled for the replica.